src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -256,16 +263,31 @@ private static Expression CreateArgument(ParameterInfo parameter, FactoryContext
             }
             else
             {
+                var nullability = NullabilityContext.Create(parameter);
+                var isOptional = parameter.HasDefaultValue || nullability.ReadState == NullabilityState.Nullable;
                 if (factoryContext.ServiceProviderIsService is IServiceProviderIsService serviceProviderIsService)
                 {
-                    // If the parameter resolves as a service then get it from services
-                    if (serviceProviderIsService.IsService(parameter.ParameterType))
+                    // If the parameter is optional
+                    if (!isOptional)
```

Comment:
?

```suggestion
                    if (isOptional)
```

We should check `IsService(parameter.ParameterType)` before getting to this point. Optionality should have no bearing whether the parameter comes from the service provider or not. It should only change how we deal with null during the request.